### DISCUSSION TITLE
How to set experiment name such that it can be some unique name instead of version_0, ... etc.

### DISCUSSION THREAD

**Discussion by dazzle-me** [Lightning-AI/pytorch-lightning#9185] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I'm currently running a lot of experiments and in order to track all of them in tensorboard I have to rename each experiment folder by hand (e.g. lightning_logs/version_0 -> lightning_logs/{unique_informative_exp_name}).
However, it would be better if I could pass the as an argument to Trainer.
I searched documentation thoroughly, but couldn't find anything related to names of experiments (I found only default_root_dir argument, however, it's not that interesting)

The question is follows :
Is there a way to set experiment name on program level? I'm using default logger.

**Comment by tshu-w** [comment, marked as answer]
You can pass Trainer a custom logger with the version specified.

```python
from pytorch_lightning.loggers import TensorBoardLogger

logger = TensorBoardLogger("default_root_dir", version="your_version", name="my_model")
trainer = Trainer(logger=logger)
```
Here is the api of [TensorBoardLogger](https://pytorch-lightning.readthedocs.io/en/stable/extensions/generated/pytorch_lightning.loggers.TensorBoardLogger.html#pytorch_lightning.loggers.TensorBoardLogger)

**Reply by dazzle-me** [reply]
Almost what I was looking for, but maybe it's even better to subdivide experiment directory into folders with model names too. 
Thanks!

**Reply by tshu-w** [reply]
> but maybe it's even better to subdivide experiment directory into folders with model names too.

You can change logger `name` to sub dir based on your model name 😄
